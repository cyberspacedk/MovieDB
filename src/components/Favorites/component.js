/* eslint-disable no-unused-vars */
/* eslint-disable react/prop-types */
import React from 'react';
import { Layout, Row, Col, Pagination, BackTop, Icon, Modal } from 'antd';
import Spinner from '../shared/StatusFields/Spinner';
import Error from '../shared/StatusFields/Error';
import Empty from '../shared/StatusFields/Empty';
import Card from '../shared/Card';

const Favorites = ({
  favoritesList,
  isLoading,
  isError,
  history,
  removeFav,
  currentPage,
  totalPage,
}) => {
  const removeFavModal = (e, id) => {
    e.stopPropagation();
    Modal.confirm({
      title: 'Do you want to delete movie from favorites?',
      onOk() {
        removeFav(id, false);
      },
      onCancel() {},
    });
  };

  return (
    <Layout>
      <Layout.Content>
        <div className="top-margin">
          <Row type="flex" gutter={16} justify="center">
            <Col span={20}>
              {isLoading && <Spinner />}
              {isError && <Error />}
              {!favoritesList.length && <Empty />}
              {favoritesList &&
                favoritesList.map(item => (
                  <Card
                    key={item.id}
                    id={item.id}
                    poster={item.poster_path}
                    title={item.original_title}
                    overview={item.overview}
                    history={history}
                    actions={[
                      <Icon
                        key="delete"
                        type="delete"
                        onClick={e => removeFavModal(e, item.id)}
                      />,
                    ]}
                  />
                ))}
            </Col>
          </Row>

          {totalPage ? (
            <Row type="flex" justify="center">
              <Col>
                <Pagination
                  defaultCurrent={1}
                  total={totalPage}
                  className="pagination"
                  defaultPageSize={20}
                />
              </Col>
            </Row>
          ) : null}
        </div>
      </Layout.Content>
      <BackTop />
    </Layout>
  );
};

export default Favorites;
