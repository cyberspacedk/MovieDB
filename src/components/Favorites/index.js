import FavoritesContainer from './container';

export default FavoritesContainer;
