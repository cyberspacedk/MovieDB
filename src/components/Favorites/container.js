/* eslint-disable react/destructuring-assignment */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Favorites from './component';
import {
  getFavoritesRequest,
  operationsFavoritesRequest,
} from '../../store/favorites/actions';
import {
  getFavoritesList,
  isError,
  isLoading,
  isEmpty,
  getTotalPages,
} from '../../store/favorites/selectors';

class FavoritesContainer extends Component {
  componentDidMount() {
    this.props.getFavoritesRequest();
  }

  render() {
    return <Favorites {...this.props} />;
  }
}

FavoritesContainer.propTypes = {
  getFavoritesRequest: PropTypes.func.isRequired,
};

const mstp = state => ({
  favoritesList: getFavoritesList(state),
  loading: isLoading(state),
  error: isError(state),
  empty: isEmpty(state),
  totalPage: getTotalPages(state),
});

const mdtp = {
  getFavoritesRequest,
  operationsFavoritesRequest,
};

export default connect(
  mstp,
  mdtp,
)(FavoritesContainer);
