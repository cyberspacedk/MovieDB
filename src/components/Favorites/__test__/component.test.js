import React from 'react';
import { shallow } from 'enzyme';
import Favorites from '../component';

describe('Component: Favorites', () => {
  const props = {
    favorites: [{ id: 1, title: 'Some title', overview: 'Some description' }],
    loading: false,
    error: false,
    empty: false,
    currentPage: 1,
    totalResults: 25,
    history: {
      push: () => {},
    },
    operationsFavoritesRequest: () => {},
    goToNextPage: jest.fn(),
    title: 'Favorites',
    removeBox: true,
  };

  it('Match its snapshot. Array provided', () => {
    const wrapper = shallow(<Favorites {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
