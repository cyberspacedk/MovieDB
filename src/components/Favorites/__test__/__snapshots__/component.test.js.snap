// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Favorites Match its snapshot. Array provided 1`] = `
<PageLayout
  array={
    Array [
      Object {
        "id": 1,
        "overview": "Some description",
        "title": "Some title",
      },
    ]
  }
  currentPage={1}
  empty={false}
  error={false}
  goToNextPage={[MockFunction]}
  history={
    Object {
      "push": [Function],
    }
  }
  loading={false}
  operations={[Function]}
  removeBox={true}
  title="Favorites"
  totalResults={25}
/>
`;
